sentry_streams/src/helper.rs

```diff
@@ -0,0 +1,26 @@
+use std::time::Instant;
+
+use pyo3::Python;
+
+pub fn traced_with_gil<F, R>(label: &str, function: F) -> R
```

Comment:
I think having the ability to trace the usage of the GIL is really good and we should keep this ability to trace, but it has to be disabled by default and we need a way to enable it.

I'd recommend having a shared readonly config object we can pass around to keep the config values this function will use to decide whether to trace the gil usage or not.

Reply:
alright

Reply:
How do we feel, instead of having each function opt-into the `traced_with_gil`, we make it so that we only emit a warning when acquiring the GIL took over a certain duration? 

That way, we require everyone to go through the `traced_with_gil` but we selectively log

Reply:
ok